src/App.js

```diff
@@ -1,16 +1,105 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm from './components/TodoComponents/TodoForm';
+import SearchButtons from './components/TodoComponents/SearchButtons';
+import './components/TodoComponents/Todo.css';
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
-  render() {
-    return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
+   constructor() {
+     super();
+     this.state = {
+       tasks:[],
+       holder: "",
+       search: "",
+       arr: [],
+     }
+   }
+
+   searchInputHandler = (event) => {
+     const {value} = event.target;
+     this.setState({
+       search: value
+     })
+   }
+
+   searchHandler = (event) => {
+     event.preventDefault();
+     this.setState({
+       arr: this.state.tasks,
+       tasks: this.state.tasks.filter(task => task.task.toLowerCase().includes(this.state.search.toLowerCase()))
+     })
+   }
+
+   showAll = (event) => {
+     event.preventDefault();
+     this.setState({
+       tasks:this.state.arr
+     })
+   }
+
+   toggleHandler = (id) => {
+     this.setState({
+       tasks:this.state.tasks.map(task => {
+         if (task.id === id) {
+           return {
+             task: task.task,
+             id: task.id,
+             completed: !task.completed
+           }
+         } else {
+           return task
+         }
+       })
+     })
+   }
+
+   deleteHandler = () => {
+     this.setState({
+       tasks: this.state.tasks.filter(task => !task.complete)
+     })
+   }
+
+   inputHandler = (event) => {
+     const {value} = event.target;
+     this.setState({
+       holder: value
+     })
+     this.state.tasks.completed === true ? event.target.classList.toggle('completed') : event.target.classList.remove('completed');
+   }
+
+   clickHandler = (event) => {
+     event.preventDefault();
+     const obj = {
+       task: this.state.holder,
+       id: Date.now(),
```

Comment:
good use of date now to get a unique id. you can also use Math.random() and there is a node module called uuid that is a great one that gives an actual unique identifier https://www.npmjs.com/package/uuid